src/libopal.c

```diff
@@ -2392,6 +2392,49 @@ short print_asm_code(asm_cmd_e cmd_list[], FILE *dest_fp)
   /// Copy NASM footer file to dest_fp
 
   /// Create strings and their lengths
+  for (int i = 0; i < strs_len; i++)
+    {
+      fprintf (dest_fp, "msg%d: DB \"", i);
+      /// Read each string character
+      for (int j = 0; j < strlen(strs[i])+1; j++)
+        {
+           ///print ASCII values for carriage returns
+           if (strs[i][j] == 13 && strs[i][j+1] == 10)
```

Comment:
This code is trying to replace `CR&NL` character which shouldn't exist inside strings -- unless [get_string_literal_lexeme()](https://github.com/mckerracher/OPaL/blob/23e919ed12caf619242eb0c6cdd22b4d4e97d5a4/src/libopal.c#L902) has a bug, where it should error out.

In this function the sequence `\n` in a string should be replaced by `", 13, 10, "`

Eg. If the program has string like this:
```
1. Hello world\n
2. Operation: \n0. Exit\n1. Add\n2. Subtract\n3. Multiply\n4. Divide\n5. Remainder\n
```
Should result in:
```
1. msg0: DB "Hello world", 13, 10, NULL
2. msg1: DB "Operation: ", 13, 10, "0. Exit", 13, 10, "1. Add", 13, 10, "2. Subtract", 13, 10, "3. Multiply", 13, 10, "4. Divide", 13, 10, "5. Remainder", 13, 10, "", NULL
```

Notice in the second example above, the `\n` after Remainder needs to be followed by an empty `""` else NASM complains.

Reply:
I filed the bug #98 for get_string_literal_lexeme() which I noticed while reviewing this code :)
I have a pull request open to fix it.

Reply:
That  makes things much easier :)